### PR TITLE
feat: switch to PDM as the package manager

### PR DESCRIPTION
**Pull request by frostming**
Signed-off-by: Frost Ming <me@frostming.com>

First, as the author of PDM, I would like to thank you for showing interest in PDM and open for a PR. In this PR I do some necessary changes to switch to PDM, including:

1. Project metadata in `pyproject.toml`
2. Dependencies moved to `pyproject.toml`, removing `requirements*.txt`
3. Update workflows according to the changes
4. Move some common tasks to PDM scripts. Update docs
5. Update the docker build scripts since it looks a bit outdated

I also found there are migration scripts to be generated. And the static files are not served in with gunicorn. These seem unrelated to this. I can do it in the next PR.

/cc @laike9m

### CODE REVIEW

**Comment by laike9m** [comment]
I didn't expect this 😂

File: docker/entrypoint.sh [hunk 1]
```diff
@@ -1,13 +1,12 @@
 #!/bin/bash
 
-set -o errexit  
-set -o pipefail  
+set -o errexit
+set -o pipefail
 set -o nounset
 
-python manage.py collectstatic --noinput
```
Comment (laike9m) [inline docker/entrypoint.sh]: these are not equivalent right?

Comment (frostming) [reply docker/entrypoint.sh]: Do you mean the removed `makemigrations` lines? Those apps don't exist any more so I think it is outdated. Anyway `python manage.py makemigrations/migrate` should handle all.

The `collectstatic` line is kept below.

File: doc/install.md [hunk 2]
```diff
@@ -42,85 +45,84 @@ SECRET_KEY = "my_key"
...
 
+Install all dependencies
+```
+pdm install
 ```
 
 Quick check
 ```
-python3 manage.py check
+pdm check
```
Comment (laike9m) [inline doc/install.md]: I might be missing something, doesn't pdm requrie a `run` command?
https://pdm.fming.dev/latest/usage/scripts/ doesn't seem to include this usage

Comment (frostming) [reply doc/install.md]: https://pdm.fming.dev/latest/usage/scripts/#:~:text=Yarn%2Dlike%20script%20shortcuts

Comment (laike9m) [reply doc/install.md]: Thanks, good to know

File: doc/install.md [hunk 3]
```diff
@@ -42,85 +45,84 @@ SECRET_KEY = "my_key"
...
 ```
 
 Initialize database
 ```
-python3 manage.py migrate
+pdm db
```
Comment (tssujt) [inline doc/install.md]: I think it may be better to use `pdm initdb`.

Comment (laike9m) [reply doc/install.md]: +1

Comment (tssujt) [reply doc/install.md]: Or `pdm dbmigrate`, like `dbshell`

**Comment by frostming** [comment]
A bit disappointed about the reaction of this PR. I am not going to do conflict resolving again. Closing it now. Thanks.

**Comment by laike9m** [comment]
I feel neodb is just another "open-source" project that only the maintaining team can actually contribute. I'll keep using it but have also given up on the idea of contributing seen how PRs are handled.

**Comment by frostming** [comment]
@alphatownsman Thank you for your kindness. But I would rather you and your team reject this PR with a firm tone so that I know I don't have to put more effort into it.

**Pull request closed without merging**